backend/kesaseteli/common/tests/test_factories.py

```diff
@@ -190,6 +190,38 @@ def test_youth_application_factory(  # noqa: C901
             assert getattr(youth_application, attribute_name) == expected_value
 
 
+@pytest.mark.django_db
+@override_settings(NEXT_PUBLIC_MOCK_FLAG=True)
+@pytest.mark.parametrize(
+    "youth_application_factory,expected_need_additional_info",
+    [
+        (AdditionalInfoRequestedYouthApplicationFactory, True),
+        (AdditionalInfoProvidedYouthApplicationFactory, True),
+        (InactiveNoNeedAdditionalInfoYouthApplicationFactory, False),
+        (InactiveNeedAdditionalInfoYouthApplicationFactory, True),
+    ],
+)
+@pytest.mark.parametrize("next_public_disable_vtj", [False, True])
+@pytest.mark.parametrize(
+    "freeze_date", ["2021-01-01", "2021-06-15", "2021-12-31", "2023-07-16"]
+)
+def test_need_additional_info(
```

Comment:
Perhaps we could also create a test `test_doesnt_need_additional_info` and use specific days that it should not apply and specific factories where it should not ask for additional info?

Reply:
`test_need_additional_info` includes parametrization to test that `InactiveNoNeedAdditionalInfoYouthApplicationFactory` returns YouthApplication objects where their `need_additional_info` is False. `test_youth_application_factory` test above this test in the same file also tests `need_additional_info` a bit but not very extensively.